### PR TITLE
Add MLIR Interpreter

### PR DESCRIPTION
**Pull request by harsh-nod**
This PR adds a torch-based MLIR interpreter that can be used for debugging IR generated by tkw. The interpreter goes through the asm line by line and maps the MLIR operations to the appropriate torch operations.

### CODE REVIEW

File: tests/tools/interpreter_test.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from ...tools.interpreter import Interpreter
+import logging
+import unittest
+from ...tests.kernel.wave.wave_gemm_test import Test as wave_gemm_test
+
+
+class Test(unittest.TestCase):
```
Comment (Hardcode84) [inline tests/tools/interpreter_test.py:7]: nti: maybe switch to pytest entirely? You won't need this `unittest.main()` as well

File: tests/tools/interpreter_test.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    def interpreter_test():
+        asm = wave_gemm_test.testGemm()
+        interpreter = Interpreter([0, 0, 0], [0, 0, 0])
+        interpreter.parse(asm)
```
Comment (Hardcode84) [inline tests/tools/interpreter_test.py:11]: should be `interpret`?

File: tests/tools/interpreter_test.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from ...tools.interpreter import Interpreter
+import logging
+import unittest
+from ...tests.kernel.wave.wave_gemm_test import Test as wave_gemm_test
```
Comment (Hardcode84) [inline tests/tools/interpreter_test.py:4]: Yeah, unfortunately, you cannot do relative imports if you are not inside the package, just copypaste gemm here?

**Review by Hardcode84 — APPROVED** [review]
LGTM, really cool!

CI is broken, but I'm fine even if it wouldn't have any test.

File: tests/tools/interpreter_test.py (hunk 3 above)
Comment (harsh-nod) [reply tests/tools/interpreter_test.py:4]: okay will do.

File: tests/tools/interpreter_test.py (hunk 2 above)
Comment (harsh-nod) [reply tests/tools/interpreter_test.py:11]: yes will fix thanks.

File: tests/tools/interpreter_test.py (hunk 1 above)
Comment (harsh-nod) [reply tests/tools/interpreter_test.py:7]: sure will remove.

**Comment by harsh-nod** [comment]
Haven't done extensive testing on this, but atleast it doesn't crash for now. Will merge it in and please feel free to modify and adapt to serve your needs.

**Pull request merged**